dotCMS/src/main/java/com/dotcms/webdav/FsDirectoryResource.java

```diff
@@ -0,0 +1,263 @@
+    }
+
+    /**
+     * Will redirect if a default page has been specified on the factory
+     *
+     * @param request
+     * @return
+     */
+    @Override
+    public String checkRedirect(Request request) {
+        if (factory.getDefaultPage() != null) {
+            return request.getAbsoluteUrl() + "/" + factory.getDefaultPage();
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public Resource createNew(String name, InputStream in, Long length, String contentType) throws IOException {
+        File dest = new File(this.getFile(), name);
+        FileOutputStream out = null;
+        try {
+            out = new FileOutputStream(dest);
+            IOUtils.copy(in, out);
+        } finally {
+            IOUtils.closeQuietly(out);
+        }
+        // todo: ignores contentType
+        return factory.resolveFile(this.host, dest);
+
+    }
+
+    @Override
+    protected void doCopy(File dest) {
+        try {
+            FileUtils.copyDirectory(this.getFile(), dest);
+        } catch (IOException ex) {
+            throw new RuntimeException("Failed to copy to:" + dest.getAbsolutePath(), ex);
+        }
+    }
+
+    @Override
+    public LockToken createAndLock(String name, LockTimeout timeout, LockInfo lockInfo) throws NotAuthorizedException {
+        File dest = new File(this.getFile(), name);
+        createEmptyFile(dest);
+        FsFileResource newRes = new FsFileResource(host, factory, dest, contentService);
+        LockResult res = newRes.lock(timeout, lockInfo);
+        return res.getLockToken();
+    }
+
+    private void createEmptyFile(File file) {
+        FileOutputStream fout = null;
+        try {
+            fout = new FileOutputStream(file);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } finally {
+            IOUtils.closeQuietly(fout);
+        }
+    }
+
+    /**
+     * Will generate a listing of the contents of this directory, unless the
+     * factory's allowDirectoryBrowsing has been set to false.
+     *
+     * If so it will just output a message saying that access has been disabled.
+     *
+     * @param out
+     * @param range
+     * @param params
+     * @param contentType
+     * @throws IOException
+     * @throws NotAuthorizedException
+     */
+    @Override
+    public void sendContent(OutputStream out, Range range, Map<String, String> params, String contentType) throws IOException, NotAuthorizedException {
+        String subpath = getFile().getCanonicalPath().substring(factory.getRoot().getCanonicalPath().length()).replace('\\', '/');
+        String uri = subpath;
+        //String uri = "/" + factory.getContextPath() + subpath;
+        XmlWriter w = new XmlWriter(out);
+        w.open("html");
+        w.open("head");
+        w.writeText(""
+                + "<script type=\"text/javascript\" language=\"javascript1.1\">\n"
+                + "    var fNewDoc = false;\n"
+                + "  </script>\n"

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "  </script>\n" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)